commons/src/main/java/fr/free/nrw/commons/upload/GPSExtractor.java

```diff
@@ -33,21 +69,67 @@ public String getCoords() {
         }
 
         if (exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE) == null) {
-            Log.d("Image", "Picture has no GPS info");
-            return null;
-        }
-        else {
+            imageCoordsExists = false;
+            Log.d(TAG, "Picture has no GPS info");
+
+            //Check what user's preference is for automatic location detection
+            boolean gpsPrefEnabled = gpsPreferenceEnabled();
+
+            if (gpsPrefEnabled) {
+                //If pref enabled, get current location
+                Location location = locationManager.getLastKnownLocation(provider);
+
+                if (location != null) {
+                    myLocationListener.onLocationChanged(location);
```

Comment:
@niedzielski Ah, okay, I think I get it - both getLastKnownLocation AND onLocationChanged should only be called once at the start, yeah? I've modified my code, do let me know if that isn't what you meant.